pandas/core/indexes/base.py

```diff
@@ -5046,14 +5046,18 @@ def get_slice_bound(self, label, side: str_t, kind) -> int:
         int
             Index of label.
         """
-        assert kind in ["loc", "getitem", None]
-
         if side not in ("left", "right"):
             raise ValueError(
                 "Invalid value for side kwarg, must be either "
                 f"'left' or 'right': {side}"
             )
 
+        if kind not in ("loc", "getitem", None):
+            raise ValueError(
```

Comment:
this should be an AssertionError  as this is an internal method (yes I know @RhysU is actually calling it), but its officially public and merely an interface from `.loc`